src/clr-addons/quick-list/quick-list.ts

```diff
@@ -4,34 +4,56 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  AfterViewChecked,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  QueryList,
+  ViewChildren,
+} from '@angular/core';
 import { CLR_BLANK_OPTION, ClrQuickListValue } from './add-option.service';
+import { ClrAddOption } from './add-option';
 
 @Component({
   selector: 'clr-quick-list',
   host: { '[class.quick-list]': 'true' },
   templateUrl: './quick-list.html',
 })
-export class ClrQuickList<T> implements OnInit {
+export class ClrQuickList<T> implements OnInit, AfterViewChecked {
   @Input('clrBlankOption') blankOption: ClrQuickListValue<T> = CLR_BLANK_OPTION;
   @Input('clrAllValues') allValues: Array<ClrQuickListValue<T>> = [this.blankOption];
   @Input('clrMandatory') mandatory: boolean = false;
   @Input('clrValues') values: Array<ClrQuickListValue<T>> = [];
   @Input('clrAddLabel') addLabel: string = 'ADD';
-  @Input('clrControlClasses') controlClasses = 'clr-col-md-10';
+  @Input('clrControlClasses') controlClasses = '';
   @Output('clrValuesChanged') valuesChanged = new EventEmitter<Array<ClrQuickListValue<T>>>();
   @Output('clrEmptyOptionAdded') emptyOptionAdded = new EventEmitter<void>();
 
+  @ViewChildren(ClrAddOption) options: QueryList<ClrAddOption<T>>;
+
+  focusId: string = null;
+
   ngOnInit(): void {
     if (this.values.length === 0 && this.mandatory) {
       this.values.push(this.blankOption);
     }
   }
 
+  ngAfterViewChecked(): void {
+    if (!!this.focusId) {
```

Comment:
this whole focusing can be moved to the option itself, as the "set selectedValue" is triggered on change.
that way you don't need to remember any id, but just focus in the method "set selectedValue" (maybe with a setTimeout to wait for everything to finish)

Reply:
i need to set the focus on 2 events
add Option
value changed 

the problem is for add option which is an event triggered from outside the option

Reply:
is set selectedValue not triggered ONCE after you click on add option? I would think you can just focus in the set selectedValue method of the option itself as marcel described. if not please describe in more detail why that doesnt work.
(if the setter is actually not called for whatever reason, you can still focus on ngAfterViewInit? )

Reply:
the add option button adds a new selection field
that selection field needs to be focused automatically
the option itsself cannot know if its the last option in the list of options in order to set the focus on itsself

Reply:
i tried many solutions, including this one, but they were not complete, or did not work at all